integration-tests/src/test/java/org/web3j/ens/EnsIT.java

```diff
@@ -3,20 +3,35 @@
 import org.junit.Test;
 
 import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.TestParameters;
 import org.web3j.protocol.http.HttpService;
 
+import static org.hamcrest.CoreMatchers.allOf;
 import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.notNullValue;
 import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeThat;
 
 public class EnsIT {
 
     @Test
     public void testEns() throws Exception {
+        HttpService httpService = new HttpService(TestParameters.TEST_RINKEBY_URL);
+        if (TestParameters.hasRinkebyCredentials()) {
+            httpService.addHeader("Authorization", TestParameters.getRinkebyAuthorization());
+        }
+        Web3j web3j = Web3j.build(httpService);
 
-        Web3j web3j = Web3j.build(new HttpService());
-        EnsResolver ensResolver = new EnsResolver(web3j);
+        assumeThat("Skipping testEns() because we are still syncing, which means we will NOT be "
+                        + "able to accurately do EnsResolver#resolve()",
+                web3j.ethSyncing().send().isSyncing(),
+                allOf(
+                        notNullValue(),
+                        is(false)
+                ));
 
+        EnsResolver ensResolver = new EnsResolver(web3j);
         assertThat(ensResolver.resolve("web3j.test"),
-                is("0x19e03255f667bdfd50a32722df860b1eeaf4d635"));
+                is("0xe7fdbbac9325508269d772982cc68103f79d9693"));
```

Comment:
http://docs.ens.domains/en/latest/introduction.html#ens-on-ethereum

web3j.test will expire every 28 days - how will this affect this test? is worth getting the permanent web3j.eth?